ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh

```diff
@@ -238,7 +238,16 @@ function filter_test_by_capability() {
     done
     echo_e2e_tags
 }
+function filter_test_by_profile() {
+    local featureset
+    featureset="$(oc get featuregate cluster -o yaml | yq '.spec.featureSet')"
```

Comment:
```suggestion
    featureset="$(oc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name'')
```
How about get the actual enabled featureGates instead(4.18+ still have the featureGate just enabled by default)? Then we could also remove the `E2E_RUN_TAGS: not @storage` for 4.17 above jobs definition.

Reply:
Great. This is what I trying to achieve, but unable to get this point. I'll update.  Thanks.

Reply:
Updated.  Appended a `sort | uniq` to the command to shorten the list.
```
$ oc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name' | wc -l
612
$ oc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name' | sort | uniq | wc -l
24
```

Reply:
Good finding! I only tried the cluster without setting `CustomNoUpgrade`, seems some specifical setting clusters have duplicated items.
```console
$ omc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name' | wc -l
      28
$ omc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name' | sort| uniq | wc -l
      28
```

Reply:
I don't have `CustomNoUpgrade` in the cluster. I think the difference is that I have a long list of upgrade history.
```
$ oc get featuregate cluster -o yaml | yq '.status.featureGates[].version' | wc -l
39
$ oc get featuregate cluster -o yaml | yq '.status.featureGates[].version' | tail -3
4.14.16
4.14.15
4.14.14
```

Reply:
Sorry, I missed one point. Even if other platforms clusters also have the `VSphereMultiVCenters` featuregate enabled, so we also need check the platform type is vsphere.
```console
$ oc version
Client Version: 4.18.0-ec.3
Kustomize Version: v5.4.2
Server Version: 4.19.0-0.nightly-2025-01-07-124551
Kubernetes Version: v1.31.3
$ oc get infrastructure cluster -oyaml|yq '.status.platformStatus.type'
AWS
$ oc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name' | sort| uniq | grep -i 'vsphere'
VSphereControlPlaneMachineSet
VSphereDriverConfiguration
VSphereHostVMGroupZonal
VSphereMultiNetworks
VSphereMultiVCenters
VSphereStaticIPs
```

Reply:
Sorry again, your update so fast, I test my fix just now. We should use(even if the platform is vcenter for only one vcenter clusters the featuregate will also be enabled ... )
```shell
# Get the number of unique vcenter servers in the failureDomains, for other platforms does not have the field
vcenters_count=$(oc get infrastructure cluster -o yaml | yq '.spec.platformSpec.vsphere.failureDomains[].server' | uniq | wc -l)
enabledgates="$(oc get featuregate cluster -o yaml | yq '.status.featureGates[].enabled[].name' | sort | uniq)"
if [[ "$vcenters_count" -ge 2 ]] && ! (grep --ignore-case --quiet 'VSphereMultiVCenters' <<< "$enabledgates") ; then
    export E2E_RUN_TAGS="not @storage and ${E2E_RUN_TAGS}"
fi
```

Reply:
Updated.  
Changed the logic a little bit, which moved the vcenters get/check inside the featuregate check.  If the featuregate is not enabled, then we can skip the command to get/check the vcenters, which should save us  one or two seconds.